aws/ecs_cluster_test.go

```diff
@@ -0,0 +1,114 @@
+	require.NoError(t, err)
+
+	parsedTagValue, parseErr1 := parseTimestampTag(formatTimestampTag(tagValue))
+	require.NoError(t, parseErr1)
+
+	parsedReturnValue, parseErr2 := parseTimestampTag(formatTimestampTag(returnedTag))
+	require.NoError(t, parseErr2)
+
+	//compare that the tags' Time values after formatting are equal
+	assert.Equal(t, parsedTagValue, parsedReturnValue)
+}
+
+// Test we can get all ECS clusters younger than < X time based on tags
+func TestCanListAllEcsClustersOlderThan24hours(t *testing.T) {
+	t.Parallel()
+
+	awsSession, err := session.NewSession(&awsgo.Config{
+		Region: awsgo.String(region),
+	})
+	require.NoError(t, err)
+
+	cluster1 := createEcsFargateCluster(t, awsSession, util.UniqueID())
+	defer deleteEcsCluster(awsSession, cluster1)
+	cluster2 := createEcsFargateCluster(t, awsSession, util.UniqueID())
+	defer deleteEcsCluster(awsSession, cluster2)
+
+	now := time.Now().UTC()
+	var olderClusterTagValue = now.Add(time.Hour * time.Duration(-48))
+	var youngerClusterTagValue = now.Add(time.Hour * time.Duration(-23))
+
+	err1 := tagEcsClusterWhenFirstSeen(awsSession, cluster1.ClusterArn, olderClusterTagValue)
+	require.NoError(t, err1)
+	err2 := tagEcsClusterWhenFirstSeen(awsSession, cluster2.ClusterArn, youngerClusterTagValue)
+	require.NoError(t, err2)
+
+	last24Hours := now.Add(time.Hour * time.Duration(-24))
+	filteredClusterArns, err := getAllEcsClustersOlderThan(awsSession, region, last24Hours)
+	require.NoError(t, err)
+
+	assert.Equal(t, 1, len(filteredClusterArns))
+}
+
+// Test we can nuke all ECS clusters older than 24hrs
+func TestCanNukeAllEcsClustersOlderThan24Hours(t *testing.T) {
+	t.Parallel()
+
+	awsSession, err := session.NewSession(&awsgo.Config{
+		Region: awsgo.String(region),
+	})
+	require.NoError(t, err)
+
+	cluster1 := createEcsFargateCluster(t, awsSession, util.UniqueID())
+	defer deleteEcsCluster(awsSession, cluster1)
+	cluster2 := createEcsFargateCluster(t, awsSession, util.UniqueID())
+	defer deleteEcsCluster(awsSession, cluster2)
+	cluster3 := createEcsFargateCluster(t, awsSession, util.UniqueID())
+	defer deleteEcsCluster(awsSession, cluster3)
+
+	now := time.Now().UTC()
+	var oldClusterTagValue1 = now.Add(time.Hour * time.Duration(-48))
+	var youngClusterTagValue = now
+	var oldClusterTagValue2 = now.Add(time.Hour * time.Duration(-27))
+
+	err1 := tagEcsClusterWhenFirstSeen(awsSession, cluster1.ClusterArn, oldClusterTagValue1)
+	require.NoError(t, err1)
+	err2 := tagEcsClusterWhenFirstSeen(awsSession, cluster2.ClusterArn, youngClusterTagValue)
+	require.NoError(t, err2)
+	err3 := tagEcsClusterWhenFirstSeen(awsSession, cluster3.ClusterArn, oldClusterTagValue2)
+	require.NoError(t, err3)
+
+	last24Hours := now.Add(time.Hour * time.Duration(-24))
+	filteredClusterArns, err := getAllEcsClustersOlderThan(awsSession, region, last24Hours)
+	require.NoError(t, err)
+
+	nukeEcsClusters(awsSession, filteredClusterArns)
+	require.NoError(t, err)
```

Comment:
```suggestion
	nukeErr := nukeEcsClusters(awsSession, filteredClusterArns)
	require.NoError(t, nukeErr)
```

Check your IDE 😁

Reply:
hahaha, no way! I installed the extensions I found! can't believe it keeps happening!

Reply:
I may have missed others, so yea, best to let the computer do the hard work 😁